xdsl/dialects/builtin.py

```diff
@@ -628,49 +635,45 @@ def shape_is_complete(self) -> bool:
         return n == len(self.data.data)
 
     @staticmethod
-    def create_dense_index(
-            type: VectorOrTensorOf[IndexType],
-            data: List[int | IntegerAttr[IndexType]]
-    ) -> DenseIntOrFPElementsAttr:
-        attr_list = [
-            IntegerAttr.from_index_int_value(d) if isinstance(d, int) else d
-            for d in data
-        ]
+    def create_dense_index(type: VectorOrTensorOf[IndexType],
+                           data: List[int]) -> DenseIntOrFPElementsAttr:
+        attr_list = [IntegerAttr.from_index_int_value(d) for d in data]
         return DenseIntOrFPElementsAttr([type, ArrayAttr(attr_list)])
 
     @staticmethod
-    def create_dense_int(
-        type: VectorOrTensorOf[IntegerType],
-        data: List[int | IntegerAttr[IntegerType]]
-    ) -> DenseIntOrFPElementsAttr:
+    def create_dense_int(type: VectorOrTensorOf[IntegerType],
+                         data: Sequence[int]) -> DenseIntOrFPElementsAttr:
         attr_list = [
-            IntegerAttr.from_params(d, type.element_type) if isinstance(
-                d, int) else d for d in data
+            IntegerAttr.from_params(d, type.element_type) for d in data
         ]
         return DenseIntOrFPElementsAttr([type, ArrayAttr(attr_list)])
 
     @staticmethod
     def create_dense_float(
             type: VectorOrTensorOf[AnyFloat],
-            data: List[int | float | AnyFloatAttr]
-    ) -> DenseIntOrFPElementsAttr:
-        data_attr = [
-            FloatAttr(float(d), type.element_type)
-            if not isinstance(d, FloatAttr) else d for d in data
-        ]
+            data: Sequence[int | float]) -> DenseIntOrFPElementsAttr:
+        data_attr = [FloatAttr(float(d), type.element_type) for d in data]
         return DenseIntOrFPElementsAttr([type, ArrayAttr(data_attr)])
 
     @staticmethod
-    def from_list(
-        type: VectorOrTensorOf[Attribute], data: List[int | AnyIntegerAttr]
-        | List[int | float | AnyFloatAttr]
-    ) -> DenseIntOrFPElementsAttr:
+    def from_list(type: VectorOrTensorOf[Attribute],
+                  data: Sequence[int | float]) -> DenseIntOrFPElementsAttr:
+
+        def type_map(el: int | float) -> int:
+            assert isinstance(el, int)
+            return el
+
         if isinstance(type.element_type, IntegerType):
-            return DenseIntOrFPElementsAttr.create_dense_int(type, data)
+            _type = cast(VectorOrTensorOf[IntegerType], type)
```

Comment:
Shouldn't this be `VectorOrTensorOf[IntegerAttr[IntegerType]]`?
Same thing for the index after.

Reply:
I didn't change this bit of the behaviour, the `create_dense_(int|float|index)` all took the type directly

Reply:
Oh my bad!